subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,262 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[publishing_overview]]
+== Publishing
+
+The vast majority of software projects build something that aims to be consumed in some way. It could be a library that other software projects use or it could be an application for end users. _Publishing_ is the process by which the thing being built is made available to consumers.
+
+In Gradle, that process looks like this:
+
+ 1. Define what to publish (_<<glossary:publication,publications>>_)
+ 2. Define where to publish it to (_<<sub:terminology_repository,repositories>>_)
+ 3. Do the publishing (via _tasks_)
+
+Each of the these steps is dependent in some way on the type of repository to which you want to publish artifacts. The two most common types are Maven-compatible and Ivy-compatible repositories, or Maven and Ivy repositories for short.
+
+NOTE: Looking for information on upload tasks and the `archives` configuration? See the <<artifact_management>> chapter.
+
+Gradle makes it easy to publish to these types of repository by providing some prepackaged infrastructure in the form of the <<publishing_maven,Maven Publish Plugin>> and the <<publishing_ivy,Ivy Publish Plugin>>. These plugins allow you to configure what to publish and perform the publishing with a minimum of effort.
```

Comment:
What's wrong with "types of repository"? Is this a US vs British English thing?

I'm a little averse to "repository types" because it can be read as "things of type 'repository'", but that shouldn't be a problem in this context. I'll switch to that if folks are more comfortable with it.

Reply:
I'm not a native speaker. So, if you say it's right I trust you.

Reply:
@eriwen What's your take on this? I ask because I'm convinced someone else brought this up a while ago, so perhaps it sounds odd to lots of people. In which case, changing it would make sense. And I can adjust my writing going forward.